### PR TITLE
Add C4 performance and shapevalidation

### PR DESCRIPTION
**Pull request by a-crate**
Also added an optional filter flag to pare down perf and shapevalidation cases to only what is needed, should make support matrixing faster

/cc @zmarano @koln67

### CODE REVIEW

File: imagetest/test_suites/networkperf/setup.go [hunk 1]
```diff
@@ -15,7 +16,10 @@ import (
 // Name is the name of the test package. It must match the directory name.
 var Name = "networkperf"
 
+var testFilter = flag.String("networkperf_test_filter", ".*", "regexp filter for storageperf test cases, only cases with a matching name will be run")
```
Comment (koln67) [inline imagetest/test_suites/networkperf/setup.go:19]: small typo: should probably say "filter for networkperf test cases"

Comment (a-crate) [reply imagetest/test_suites/networkperf/setup.go]: Woops, yeah.

**Comment by a-crate** [comment]
C4 shapevalidation case names were colliding. Should be ok now.

**Comment by a-crate** [comment]
I'll re-request review from you when I have the presubmits for this working..

**Comment by a-crate** [comment]
Ok should be ready now

**Pull request merged**